extensions/PhabBugz/Extension.pm

```diff
@@ -54,6 +58,34 @@ sub webservice {
   $args->{dispatch}->{PhabBugz} = "Bugzilla::Extension::PhabBugz::WebService";
 }
 
+sub object_end_of_update {
+  my ($self,   $args)    = @_;
+  my ($object, $changes) = @$args{qw(object changes)};
+  my $params = Bugzilla->params;
+
+  next
+    if !$params->{phabricator_enabled}
+    || !$object->isa('Bugzilla::User')
+    || !$changes
+    || !$changes->{disabledtext};
+
+  try {
+    my $phab_user
+      = Bugzilla::Extension::PhabBugz::User->new_from_query({ids => [$object->id]});
+    return if !$phab_user;
+    if ($object->is_enabled) {
+      $phab_user->enable_user;
+    }
+    else {
+      $phab_user->disable_user;
+    }
+  }
+  catch {
+    WARN('ERROR: Error updating Phabricator user status for Bugzilla user '
+        . $object->login);
```

Comment:
I suppose if this ever fails, we'll probably miss the warning, but we'll still once-in-a-while ensure that Phabricator's "disabled users" are up-to-date with Bugzilla, is that right?

Since this (guessed 😄) situation is still strictly better than today's situation, that sounds good to me 👍

Reply:
Yeah. It is not optimal in that it is best effort and doesn't retry on failure. And we would need to know the error was occurring by either watch the logs or noticing a user still is enabled in Phab in future. Not sure if there is a security risk in having a user not be disabled in Phab and disabled in BMO. The user would remain logged into Phab for a long time even if not logged in BMO. We could instead send an email to bmo-admins instead of just logging a warning. Then we could address it manually. I weighed this against just out right failing when someone disables the user in BMO and did not want to hinder that process since #servicedesk handles that,

Reply:
If it's not too tricky, then sending an email to `bmo-admins` (or otherwise triggering an alert notification in our monitoring infrastructure) sounds great!
I'm happy to land that in a separate patch if you'd prefer to land this one sooner, especially considering how this is a drastic improvement on the status quo.